src/pages/workspace/WorkspaceInvitePage.js

```diff
@@ -65,93 +66,106 @@ const defaultProps = {
         // Exclude any expensify emails or valid policy members from the invite options
         const memberEmailsToExclude = [...CONST.EXPENSIFY_EMAILS];
-        _.each(this.props.policyMembers, (policyMember, accountID) => {
+        _.each(props.policyMembers, (policyMember, accountID) => {
             // Policy members that are pending delete or have errors are not valid and we should show them in the invite options (don't exclude them).
             if (policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || !_.isEmpty(policyMember.errors)) {
                 return;
             }
-            const memberEmail = lodashGet(this.props.personalDetails, `[${accountID}].login`);
+            const memberEmail = lodashGet(props.personalDetails, `[${accountID}].login`);
             if (!memberEmail) {
                 return;
             }
             memberEmailsToExclude.push(memberEmail);
         });
         return memberEmailsToExclude;
-    }
+    }, [props.personalDetails, props.policyMembers]);
+
+    const {personalDetails, userToInvite} = useMemo(
+        () => OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, '', excludedUsers),
+        [props.personalDetails, props.betas, excludedUsers],
+    );
+
+    const updateOptionsWithSearchTerm = useCallback(
+        (newSearchTerm = '') => {
+            // Update selectedOptions with the latest personalDetails and policyMembers information
+            const detailsMap = {};
+            _.forEach(personalDetails, (detail) => (detailsMap[detail.login] = detail));
+
+            const newSelectedOptions = [];
+            _.forEach(selectedOptions, (option) => {
+                newSelectedOptions.push(_.has(detailsMap, option.login) ? detailsMap[option.login] : option);
+            });
+            setSearchTerm(newSearchTerm);
+            setSelectedOptions(newSelectedOptions);
+        },
+        [personalDetails, selectedOptions],
+    );
+
+    useEffect(() => {
+        clearErrors();
+        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(props.policyMembers, props.personalDetails);
+        Policy.openWorkspaceInvitePage(props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
+    }, [clearErrors, props.route.params.policyID, props.policyMembers, props.personalDetails]);
+
+    useEffect(() => {
+        if (!_.isEqual(prevPersonalDetails, props.personalDetails) || !_.isEqual(prevPolicyMembers, props.policyMembers)) {
+            updateOptionsWithSearchTerm(searchTerm);
+        }
+
+        const isReconnecting = wasOffline && !isOffline;
+        if (!isReconnecting) {
+            return;
+        }
+
+        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(props.policyMembers, props.personalDetails);
+        Policy.openWorkspaceInvitePage(props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
```

Comment:
Why do we need to put this all in this block, we can simplify and break this down into smaller blocks and make it reusable

Reply:
Can you please explain to me like how to break it into smaller chunks?

Reply:
This block is a refactored version of the `componentDidupdate`.  It was a lifecycle method so we needed it to keep it together and handle it based if checks now that's not the case. So why need it keep it together?

I've suggested this already [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1688145321114179?thread_ts=1688112056.956949&cid=C01GTK53T8Q) . 
Please learn how to do migration and refer to this official doc https://react.dev/reference/react/Component#alternatives and other closed migration PRs